src/main/java/calculator/NumberExtractor.java

```diff
@@ -0,0 +1,45 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class NumberExtractor {
+
+    private String input;
+    private String divider = "[,:]";
+
+    private final Matcher MATCHER;
+
+    public NumberExtractor(String input) {
+        this.input = input;
+        Pattern pattern = Pattern.compile("//(.)\n(.*)");
+        this.MATCHER = pattern.matcher(input);
+    }
+
+    public List<Integer> extractNumbers() {
+
+        checkExtractorMode();
+
+        return Arrays.stream(this.input.split(divider))
+                .map(number -> {
+                    try {
+                        int intNumber = Integer.parseInt(number);
+                        if (intNumber < 0) throw new RuntimeException();
+                        return intNumber;
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                })
+                .collect(Collectors.toList());
+    }
```

Comment:
이 메소드는 아래의 여러 역할을 수행하고 있어요.
- input String을 divider를 통해 나누어 준다.
- 나누어진 각각의 number String 값을 int로 변환한다.
- 변환하는 과정에서 0보다 작은 경우 예외를 발생시킨다.

단일 책임 원칙에 맞게 최소한의 역할만 수행하도록 변경해보시는건 어떨까요?
굳이 과하게 나눌 필요는 없지만, 연습이라고 생각하고 나눠보시는 것도 괜찮을 것 같습니다.

```suggestion
    public List<Integer> extractNumbers() {

        checkExtractorMode();

        return Arrays.stream(this.input.split(divider))
                .map(number -> convertNumber(number))
                .collect(Collectors.toList());
    }
    
    private int convertNumber(String text) {
        try {
            int intNumber = Integer.parseInt(text);
            if (intNumber < 0) throw new RuntimeException();
            return intNumber;
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
```